homeassistant/components/coolmaster/config_flow.py

```diff
@@ -0,0 +1,44 @@
+"""Config flow to configure Coolmaster."""
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+
+from .const import AVAILABLE_MODES, CONF_SUPPORTED_MODES, DEFAULT_PORT, DOMAIN
+
+MODES_SCHEMA = {vol.Required(mode): bool for mode in AVAILABLE_MODES}
+HOST_SCHEMA = {
+    vol.Required(CONF_HOST): str,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+}
+
+DATA_SCHEMA = vol.Schema({**HOST_SCHEMA, **MODES_SCHEMA})
+
+
+class CoolmasterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Coolmaster config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def _async_get_entry(self, data):
+        supported_modes = [
+            key for (key, value) in data.items() if key in AVAILABLE_MODES and value
+        ]
+        return self.async_create_entry(
+            title=data[CONF_HOST],
+            data={
+                CONF_HOST: data[CONF_HOST],
+                CONF_PORT: data[CONF_PORT],
+                CONF_SUPPORTED_MODES: supported_modes,
+            },
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            return self._async_get_entry(user_input)
```

Comment:
You need to validate that the input works.

Reply:
I added validation, but when showing an error the submit button is then outside the view. Not sure how to resolve that.

Reply:
screenshot ?

Reply:
Before:
<img width="499" alt="before" src="https://user-images.githubusercontent.com/2159693/67260915-b3cfcf80-f4a6-11e9-837e-2d1e0fcaa193.png">
After:
<img width="517" alt="after" src="https://user-images.githubusercontent.com/2159693/67260919-b7635680-f4a6-11e9-9397-4f1779dc4e9d.png">